src/main/java/report_screens/ReportScreen.java

```diff
@@ -88,8 +97,12 @@ public void actionPerformed(ActionEvent evt) {
 
 
             }
+
+            this.dispose();
         } else if(evt.getSource() == cancel){
```

Comment:
I think evt.getSource() returns the thing that created the event, in this case a JButton. You should change this to evt.getActionCommand() to be safe and probably change it to .equals() instead of ==